File: 250325/G or H 2/G-or-H-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input.shift());
const people = Array(101).fill('E');
for (let i = 0; i < n; i++) {
    const [posStr, letter] = input[i].split(' ');
    const pos = Number(posStr);
    people[pos] = letter
}
let answer = 0
for(let i = 0; i < people.length; i++){
    let countG = 0;
    let countH = 0;
    let max = 0;
    if(people[i] !== 'E'){
        if(people[i] === 'G'){
            countG++
        }else {
            countH++
        }
        for(let j = i; j < people.length; j++){
            if(people[j] !== 'E'){
                if(people[j] === 'G' && i !== j){
                    countG++
                }else if(people[j] === 'H' && i !== j) {
                    countH++
                }
                if((countG !== 0 && countH === 0) || (countG === 0 && countH !==0) || (countG === countH)){
                    max = Math.max(max, j-i);
                }
            }

            
        }
    }
    answer = Math.max(answer, max)
}

console.log(answer)
// Please Write your code here.

File: 250808/G or H 2/G-or-H-2.py
n = int(input())
people = [tuple(input().split()) for _ in range(n)]
pos = [int(p[0]) for p in people]
alpha = [p[1] for p in people]
arr = ['T'] * 101

for i in range(n):
    if(alpha[i] == 'G'):
        arr[pos[i]] = 'G'
    elif(alpha[i] == 'H'):
        arr[pos[i]] = 'H'

max_len = 0
for i in range(101):
    for j in range(i+1, 101):
        if arr[i] == 'T' or arr[j] == 'T':
            continue
        
        cnt_g = 0
        cnt_h = 0

        for k in range(i, j + 1):
            if arr[k] == 'G':
                cnt_g+=1
            if arr[k] == 'H':
                cnt_h +=1
        if cnt_g == cnt_h:
            max_len = max(max_len, j-i)
print(max_len)
